src/editor/contentState/selectionCtrl.js

```diff
@@ -0,0 +1,43 @@
+const selectionCtrl = ContentState => {
+  ContentState.prototype.tableCellHandler = function (event) {
+    const { start, end } = this.cursor
+    const startBlock = this.getBlock(start.key)
+    const endBlock = this.getBlock(end.key)
+    const { type: startType } = startBlock
+    const { type: endType } = endBlock
+    if (/th|td/.test(startType) && /th|td/.test(endType)) {
+      event.preventDefault()
+      event.stopPropagation()
+      if (start.key === end.key) {
+        const { text, key } = startBlock
+        this.cursor = {
+          start: { key, offset: 0 },
+          end: { key, offset: text.length }
+        }
+      } else {
+        // Get table from start block
+        //   table -> thead or tbody -> tr -> th or td (cell)
+        const table = this.getParent(this.getParent(this.getParent(startBlock)))
+        if (!table || table.type !== 'table') {
+          console.error('No table found or invalid type.')
+          return
+        }
+        // Get last table block
+        const lastTableBlock = this.getLastChild(this.getLastChild(this.getLastChild(table)))
```

Comment:
maybe you need `lastInDescendant` and `firstInDescendant` to find the first cell and last cell?

Reply:
:+1: